unit-tests/live/options/test-timestamp-domain.py

```diff
@@ -0,0 +1,117 @@
+# License: Apache 2.0. See LICENSE file in root directory.
+# Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+# test:device D400*
+
+import time
+import pyrealsense2 as rs
+
+from rspy import test
+
+
+def get_connected_device():
+    """
+    Get first connected device
+    :return: connected device
+    """
+    dev = None
+    try:
+        context = rs.context()
+        dev = context.devices[0]
+    except Exception as ex:
+        print('Exception: Failed to find connected device. ', str(ex))
+    return dev
+
+
+def close_resources(sensor):
+    """
+    Stop and Close sensor.
+    :param: sensor of device
+    """
+    if len(sensor.get_active_streams()) > 0:
+        sensor.stop()
+        sensor.close()
+
+
+def start_sensor_test(sensor, global_time_enabled: int):
+    """
+    Perform depth sensor test according given global time
+    :global_time_enabled int: 1 - time is enabled, 0 - time disabled
+    """
+
+    if global_time_enabled not in [0, 1]:
+        raise ValueError(f'Invalid parameter in start depth test: {global_time_enabled}. Only 1 or 0 are valid.')
+
+    try:
+        sensor.set_option(rs.option.global_time_enabled, global_time_enabled)
+        time.sleep(2)
+
+        frame = frame_queue.wait_for_frame()
+
+        if global_time_enabled == 0 and sensor.get_option(rs.option.global_time_enabled) == 0:
```

Comment:
How is this instead on lines 51-56?
```
expected_ts_domain = rs.timestamp_domain.hardware_clock if global_time_enabled else rs.timestamp_domain.global_time

test.check_equal(sensor.get_option(rs.option.global_time_enabled), global_time_enabled)
test.check_equal(frame.get_frame_timestamp_domain(), expected_ts_domain)
```

This check (and fail the test if not) the option was set correctly and the metadata value is correct.

Reply:
Could be that I mixed the values, check what is needed.
I wanted to offer the flow difference :)

Reply:
Yes, your code is more elegantic.